src/ShopBundle/objects/WebModules/ViewMyOrderDetails/MTShopViewMyOrderDetails.php

```diff
@@ -85,24 +85,39 @@ public function Accept(
             'order' => null,
         );
 
-        if (false === $viewOrderDetailHandler->orderIdBelongsToUser($orderIdRequested, $extranetUserId)) {
-            // this is not our order. access denied
-            $viewData['error'] = true;
-            $viewData['errorCode'] = 'notMyOrderError';
-        }
+        $order = $this->getOrder($orderIdRequested);
 
-        $order = $this->getDbAdapter()->getOrder($orderIdRequested);
         if (null === $order) {
             // this is not our order. access denied
             $viewData['error'] = true;
             $viewData['errorCode'] = 'orderNotFoundError';
+        } else {
+            if (null === $extranetUserId || false === $viewOrderDetailHandler->orderIdBelongsToUser($orderIdRequested, $extranetUserId)) {
+                // this is not our order. access denied
+                $viewData['error'] = true;
+                $viewData['errorCode'] = 'notMyOrderError';
```

Comment:
This error code is defined somewhere?

Reply:
I'll finish the PR as @pixeljunkie currently doesn't find the time.

These lines were only moved from above, so I'm quite sure the error code does already exist.